.github/workflows/test.yml

```diff
@@ -0,0 +1,421 @@
+        uses: actions/checkout@v2
+        with:
+          repository: aiplan4eu/up-pyperplan
+          path: up-pyperplan
+          ref: ${{env.up_pyperplan_commit}}
+
+      - name: Install up-pyperplan
+        run: python3 -m pip install up-pyperplan/
+
+      - name: Setup java for ENHSP
+        uses: actions/setup-java@v2
+        with:
+          distribution: "microsoft"
+          java-version: "17"
+
+      - name: Download and install ENHSP
+        run: |
+          git clone https://gitlab.com/enricos83/ENHSP-Public.git
+          cd ENHSP-Public; git checkout enhsp20-0.9.7; bash ./compile; cd ..
+          bash -c "mkdir .planners; mv ENHSP-Public/enhsp-dist .planners/enhsp-20; rm -rf ENHSP-Public"
+
+      - name: Run tests
+        run: bash run_tests.sh
+
+  run-tests-macos:
+    runs-on: macos-latest
+    needs: [
+        basic-check,
+        check-protobuf,
+      ]
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Set up Python
+        uses: actions/setup-python@v1
+        with:
+          python-version: ${{inputs.python-version}}
+
+      - name: Install Python dependencies
+        run: python3 -m pip install -r requirements.txt
+
+      - name: Install Development Python dependencies
+        run: python3 -m pip install -r dev-requirements.txt
+
+      - name: Install tarski
+        run: |
+          python3 -m pip install tarski[arithmetic]
+      #          brew install gringo -- Commented because not working on github actions
+      - name: Checkout up-tamer
+        uses: actions/checkout@v2
+        with:
+          repository: aiplan4eu/up-tamer
+          path: up-tamer
+          ref: ${{env.up_tamer_commit}}
+
+      - name: Install up-tamer
+        run: python3 -m pip install up-tamer/
+
+      - name: Checkout up-pyperplan
+        uses: actions/checkout@v2
+        with:
+          repository: aiplan4eu/up-pyperplan
+          path: up-pyperplan
+          ref: ${{env.up_pyperplan_commit}}
+
+      - name: Install up-pyperplan
+        run: python3 -m pip install up-pyperplan/
+
+      - name: Setup java for ENHSP
+        uses: actions/setup-java@v2
+        with:
+          distribution: "microsoft"
+          java-version: "17"
+
+      - name: Download and install ENHSP
+        run: |
+          git clone https://gitlab.com/enricos83/ENHSP-Public.git
+          cd ENHSP-Public; git checkout enhsp20-0.9.7; bash ./compile; cd ..
+          bash -c "mkdir .planners; mv ENHSP-Public/enhsp-dist .planners/enhsp-20; rm -rf ENHSP-Public"
+
+      - name: Run tests
+        run: bash run_tests.sh
+
+  check-protobuf:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Set up Python
+        uses: actions/setup-python@v1
+        with:
+          python-version: ${{inputs.python-version}}
+
+      - name: Install Protobuf compiler
+        run: |
+          python3 -m pip install grpcio-tools==1.54.2
```

Comment:
Hi @arbimo! I updated the grpcio-tools version from `1.48.1` to `1.54.2` because `1.48.1` does not contain wheels for Python 3.11 and we want to run all the tests for all the supported version (Python 3.8, 3.9, 3.10, 3.11) before every new release. Do you see any issues?

Reply:
No problem, fixing the version is only a matter of making the process reproducible. There should not be any down side of upgrading the version.